misc/python/materialize/cli/mzconduct.py

```diff
@@ -806,251 +807,190 @@ def run(self, comp: Composition, workflow: Workflow) -> None:
+
+    :param chaos:          List containing types of chaos to add. If not provided,
+                           will default to 'default_chaos'.
+    :param services:       List of target Docker services for chaos. If not provided,
+                           will default to all running Docker services.
+    :param other_service:  Chaos will be randomly added to Docker services as long as
+                           'other_service' is running. If not provided, chaos will be
+                           added forever.
+    """
+
+    default_chaos = [
+        "pause",
+        "stop",
+        "kill",
+        "delay",
+        "rate",
+        "loss",
+        "duplicate",
+        "corrupt",
+    ]
 
     def __init__(
-        self,
-        service: str,
-        other_service: str,
-        duration: int,
-        run_cmd: str,
-        stop_cmd: str,
-        run_time: int,
-        stop_time: int,
+        self, chaos: List[str] = [], services: List[str] = [], other_service: str = "",
     ):
-        if (
-            other_service == ""
-            and duration < 0
-            or other_service != ""
-            and duration >= 0
-        ):
-            raise BadSpec(
-                f"Need to specify either 'other_service' or 'duration' to run chaos Docker workflow step"
-            )
-        self._service = service
+        self._chaos = chaos
+        self._services = services
         self._other_service = other_service
-        self._duration = duration
-        self._run_cmd = run_cmd
-        self._stop_cmd = stop_cmd
-        self._run_time = run_time
-        self._stop_time = stop_time
 
-    def run(self, comp: Composition, workflow: Workflow) -> None:
-        container_id = comp.get_container_id(self._service)
-        say(
-            f"{self._stop_cmd} and {self._run_cmd} {container_id}: running for {self._run_time} seconds, \
-             stopping for {self._stop_time} seconds"
-        )
-        if self._other_service != "":
-            other_container_id = comp.get_container_id(
-                self._other_service, running=True
-            )
-            while comp.docker_container_is_running(other_container_id):
-                self.stop_and_start(container_id)
-        elif self._duration >= 0:
-            self.run_for_duration(container_id)
-
-    def run_for_duration(self, container_id: str) -> None:
-        ui.progress(f"running for {self._duration} ", "C")
-        for _ in ui.timeout_loop(self._duration):
-            self.stop_and_start(container_id)
-        ui.progress(finish=True)
-
-    def stop_and_start(self, container_id: str) -> None:
-        try:
-            spawn.runv(["docker", self._stop_cmd, container_id])
-        except subprocess.CalledProcessError as e:
-            raise Failed(f"Unable to {self._stop_cmd} container {container_id}: {e}")
-        time.sleep(self._stop_time)
+    def get_running_docker_processes(self, running: bool = False) -> str:
```

Comment:
since this doesn't use `self` it could be an `@staticmethod` or just a free function.

the name also might make more sense as `get_docker_processes` since it accepts `running` as a parameter.